### ISSUE TITLE
Float values are not parsed as floats

### ISSUE THREAD

**Issue by dburov190** [JuliaIO/ConfParser.jl#22]
The parser reads values like "1e-3" as `String`s rather than `Float64` variables, at least in the case of "INI" files.

**Comment by jonathanBieler** [comment]
I think everything is parsed as String. I think in general guessing the proper type and parsing it is quite complicated. For now you'll have to parse your parameters yourself (`parse(Float64, x)`).

**Comment by dburov190** [comment]
You're right about everything being parsed as a string, I didn't notice that... and yes, parsing is quite involved, but thanks to the standard `parse` function you mentioned it's not that hard. For example, is it possible to add an optional default (which would be helpful in and out of itself, in case there's no key you're trying to retrieve) that would be used to infer the type? something like
`parse(typeof(default), x)`
?

**Comment by pixel27** [comment]
Isn't this what retrieve(s::ConfParse, block::String, key::String, t::Type) is for?  It retrieves the key then parses it as a Float32?

**Comment by dburov190** [comment]
@pixel27 Maybe it is, but I'd have no idea because it's not documented, not even in the simplest "examples" way... see #21 (:

**Comment by jonathanBieler** [comment]
I'll try to setup the documentation.
